Source/AtomicEditor/Editors/ResourceEditor.cpp

```diff
@@ -42,11 +42,15 @@ class EditorTabLayout: public TBLayout
         if (editor_->HasUnsavedModifications())
         {
             TBMessageWindow *msg_win = new TBMessageWindow(this, TBIDC("unsaved_modifications_dialog"));
-            TBMessageWindowSettings settings(TB_MSG_OK_CANCEL, TBID(uint32(0)));
+            TBMessageWindowSettings settings(TB_MSG_NONE, TBID(uint32(0)));
             settings.dimmer = true;
             settings.styling = true;
             String windowString = Atomic::ToString("%s has unsaved modifications.\nDo you wish to discard them and close?", GetFileNameAndExtension(editor_->GetFullPath()).CString());
             msg_win->Show("Unsaved Modifications",  windowString.CString(), &settings, 640, 360);
+            TBButton *btn = new TBButton();
+            msg_win->AddButton("dont_save", false, msg_win->GetWidgetByIDAndType<TBLayout>(5));
```

Comment:
GetWidgetByIDAndType<TBLayout>(5)

Kind of magical, we're now using internal id's in external code, and there are enough of them that this gets confusing